### PR TITLE
Setup Typescript

### PR DESCRIPTION
**Pull request by synzen**
I put ES2020 as the target since we can use all these neat new shiny features https://alligator.io/js/es2020/ ;o

### CODE REVIEW

**Comment by phaseharry** [comment]
Does the features work in node yet? Im still waiting for `import` to make it lul

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
+    // "checkJs": true,                       /* Report errors in .js files. */
+    // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
+    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+    // "declarationMap": true,                /* Generates a sourcemap for each corresponding '.d.ts' file. */
+    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
+    // "outFile": "./",                       /* Concatenate and emit output to single file. */
+    "outDir": "./build",                      /* Redirect output structure to the directory. */
+    "rootDir": "./src",                       /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
+    // "composite": true,                     /* Enable project compilation */
```
Comment (phaseharry) [inline tsconfig.json]: Assuming this is a template for online and you just committed out non-essentials

Comment (synzen) [reply tsconfig.json]: yea it was using `tsc --init` and it generates a tsconfig file which lets you comment out stuff in a json file which is kinda weird but TS understands it

**Comment by synzen** [comment]
while using TS with node, indeedly do! `import`/`export` also works with TS in node

**Pull request merged**